sdk/core/azure-core/azure/core/pipeline/policies/redirect.py

```diff
@@ -141,7 +141,7 @@ def increment(self, settings, response, redirect_location):
             response.http_request.method = 'GET'
         for non_redirect_header in self._remove_headers_on_redirect:
             response.http_request.headers.pop(non_redirect_header, None)
-        return settings['redirects'] > 0 or not settings['allow']
+        return settings['redirects'] > 0
```

Comment:
Why move the check out of increment method?

Reply:
Two reasons: 
1. The check for settings['allow'] is not getting executed here. When it is configured with no_redirects the 'redirects' in the settings still defaults to 30 attempts. So it will continue to make redirect attempts before evaluating if redirects are allowed.
2. If 'allow' is checked in the increment method and is False, the method will return False which raises a TooManyRedirectsError in the send method instead of returning the response.